src/sim.rs

```diff
@@ -906,6 +912,79 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn elapsed_time_across_crashes() -> Result {
```

Comment:
This is a really great test, but I'd personally be inclined to keep it simpler (at the expense of coverage). I don't think showing host2 is unaffected by the crash is that helpful, nor do I think showing clock_1 falls behind is that helpful. Really, this could just be one host and the assertion on L974.

Reply:
I understand you probably wrote the rest of it for your own learning/exploration of current behavior, but part of my inclination is based on what happens over time if you don't "spring clean" the tests. We land up with a bunch of ~100 LOC tests that really could be 20 LOC, which makes everything that much harder to grok.

This is not a review blocker, as we can always clean it later.